test/unit/middlewares/articlesMiddleware.test.js

```diff
@@ -262,4 +299,159 @@ describe('API endpoint: /api/articles (Middleware test)', () => {
+
+  describe('GET: /api/v1/articles/:slug', () => {
+    it('Should return an error if the article id is a string', (done) => {
+      chai
+        .request(app)
+        .get('/api/v1/articles/uyo')
+        .set({ Authorization: `Bearer ${dummyUser.token}` })
+        .end((err, res) => {
+          expect(res).to.have.status(STATUS.NOT_FOUND);
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.haveOwnProperty('code').to.equal(STATUS.NOT_FOUND);
+          expect(res.body.message).to.be.equal('no article with slug: uyo found');
+          done();
+        });
+    });
+  });
+
+  describe('GET: /api/v1/articles/:slug', () => {
+    it('Should return an error if article id is less than 0', (done) => {
+      chai
+        .request(app)
+        .get(`/api/v1/articles/${-4}`)
+        .set({ Authorization: `Bearer ${dummyUser.token}` })
+        .end((err, res) => {
+          expect(res).to.have.status(STATUS.NOT_FOUND);
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.haveOwnProperty('code').to.equal(STATUS.NOT_FOUND);
+          expect(res.body.message).to.be.equal('no article with slug: -4 found');
+          done();
+        });
+    });
+  });
+
+  describe('PUT: /api/v1/articles/:articleId', () => {
+    it('Should return an error if no article id is provided', (done) => {
+      chai
+        .request(app)
+        .put(`/api/v1/articles/${'   '}`)
+        .set({ Authorization: `Bearer ${dummyUser.token}` })
+        .end((err, res) => {
+          expect(res).to.have.status(STATUS.NOT_FOUND);
+
+          done();
+        });
+    });
+  });
+
+  describe('PUT: /api/v1/articles/:articleId', () => {
+    it('Should return an error if the article id is a string', (done) => {
+      chai
+        .request(app)
+        .put('/api/v1/articles/uyo')
+        .set({ Authorization: `Bearer ${dummyUser.token}` })
+        .end((err, res) => {
+          expect(res).to.have.status(STATUS.BAD_REQUEST);
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.haveOwnProperty('code').to.equal(STATUS.BAD_REQUEST);
+          expect(res.body.message).to.be.equal('article Id is not a number');
+          done();
+        });
+    });
+  });
+
+  describe('PUT: /api/v1/articles/:articleId', () => {
+    it('Should return an error if article id is less than 0', (done) => {
+      chai
+        .request(app)
+        .put(`/api/v1/articles/${-4}`)
+        .set({ Authorization: `Bearer ${dummyUser.token}` })
+        .end((err, res) => {
+          expect(res).to.have.status(STATUS.BAD_REQUEST);
+          expect(res.body).to.be.an('object');
+          expect(res.body).to.haveOwnProperty('code').to.equal(STATUS.BAD_REQUEST);
+          expect(res.body.message).to.be.equal('article Id is not valid');
+          done();
+        });
+    });
+  });
+
+  describe('PUT: /api/v1/articles/:articleId', () => {
+    it('Should return an error if authorId is incorrect', (done) => {
```

Comment:
#### Nitpick
- AuthorId or ArticleId ?

Reply:
I'm sorry for all these details errors. Stress and pressure is the cause. I'll attend to them all